why we write (#13345)
gatsbyjs/gatsby: docs/blog/2019-04-19-gatsby-why-we-write/index.md

In the docs, things are structured hierarchically, so usually content is mostly adjacent to related content on the left sidebar.

But outside the docs, that's far from the case. We have blog posts, [using-\* example sites / guides](https://github.com/gatsbyjs/gatsby/tree/master/examples), [starters](/starters/), conference talks, podcast appearances, tweets, and whatever Gatsby community members dream up and publish!

**It's a constant challenge to organize these in ways that are useful and obvious for the end-user**. We're always attacking this in five or ten different ways -- discoverability via search, cross-linking, hierarchy, navbars and breadcrumbs, turning conference talks into blog posts and blog posts into docs. 

Our goal is to make it easy for the user to gather relevant context, know where to look when they get stuck, and feel confident that they can make it if they keep on reading, learning, and trying.

## Principle 6: make it easy for folks to become engrossed with Gatsby

> Gatsby is the future of the web. Our goal is to let people dive _as far deep into it as they want, and encourage them by providing them more and more to discover._

Reviewer: > Gatsby is the future of the web.

This is asserted again towards the end, which seems fine...but I wonder if we could differentiate the two a bit. It could lead with something like:

> Building sites with Gatsby is the way of the future.

Author: Good point! Changed.

---
title: Gatsby: Why We Write
date: 2019-04-19
author: Sam Bhagwat

> excerpt: "Why does the Gatsby team and community write so much? Six principles that explain our logorrhea."

Reviewer: > logorrhea

I will admit I had to look this up, and it was still a bit cryptic to me: "a tendency to extreme loquacity", in addition to referencing a communication disorder. Is there a more approachable word we can use? How about "our approach to content"?

Reviewer 2: How about _love for writing_? Introduces a more positive tone. I do love _logorrhea_ (it's a [winning spelling bee word](https://en.wikipedia.org/wiki/72nd_Scripps_National_Spelling_Bee)!).

Author: That seems like a good in between spot!

---

If you're a Gatsby community member, you may have noticed something: we write _a lot_.

> Community members are often surprised by the comprehensiveness of our [documentation](/docs/), and we publish new article on this blog multiple times per week, from both the core team and the community.

Reviewer: ```suggestion
Community members are often surprised by the comprehensiveness of our [documentation](/docs/). We publish new articles on this blog multiple times per week, from both the core team and the community.
```

https://twitter.com/mattconvente/status/1099706762897342465?s=19

Writing isn't something we just do because we love it (although we do). Writing is core to who we are as a project, a company, and a community.

> Writing is core to the success we've enjoyed so far, and it's core to our plans to become the default way to build on the web.

Reviewer: > it's core to our plans to become the default way to build on the web

This came off a little too strong to me. Can we soften it with "hopes"? "dreams"?

Reviewer 3: Contrary opinion here--I like pieces that have a bit of a stronger take! That _is_ our plan--so I personally am a fan of this!

Author: Good point / counterpoint! I'll use "strategy" as that makes it clear this is our ambition.

> ### Principle 1: We write

Reviewer: ```suggestion
## Principle 1: We write
```

> Some Inkteam members who have been regularly writing (speaking, appearing on podcasts, streaming): [Jason](/contributors/jason-lengstorf/), [Marcy[(https://egghead.io/podcasts/heading-gatsby-s-learning-experience-and-bridging-gaps-with-marcy-sutton), [Amberley](/contributors/amberley-romo/), [Linda](/contributors/linda-watkins/), [Dustin](/contributors/dustin-schau/), [Shannon](/contributors/shannon-soper/), [Jim](/contributors/jim-ettig/), [Marisa](/contributors/marisa-morby/), [Preston](/contributors/preston-so/), [Mikhail](/contributors/mikhail-novikov/), [Sid](/contributors/sidhartha-chatterjee/), [Andrew](https://www.gatsbyjs.com/gatsby-days-preview-andrew/), [Mike](/contributors/mike-allanson/), [John](/docs/mdx/), [me](/contributors/sam-bhagwat/), and

Reviewer: With #13346 I will have a contributor page!

[Kyle](/contributors/kyle-mathews/),

And it's not just the Inkteam. The documentation, while curated by the Inkteam, has largely been authored by enthusiastic community members, along with about half of our blog posts.

Off this blog, on Twitter and the wider web, every day brings four or five community members writing about their experience.

Why do we write so much?

The answer:

> ### Principle 2: content acts as a guide unlocking the power of Gatsby.

Reviewer: ```suggestion
## Principle 2: content acts as a guide unlocking the power of Gatsby.
```

Gatsby is so powerful, because of its core _and the ecosystem_ — not just plugins and themes, but React, Webpack and Babel, the entire npm ecosystem, and modern Javascript in general.

You can do almost anything you want with Gatsby, you just have to:

_(1) know why & that it's possible_

_(2) know how to do it_

Roughly speaking, there are two sides of content at Gatsby — marketing content which focuses on (1), and learning content which focuses on (2).

An example of (1) might be Linda's [How To Talk About Gatsby To Your Clients and Team](https://www.gatsbyjs.org/blog/2019-03-07-sell-gatsby-to-clients/).

An example of (2) might be Jason's [Learn With Jason](https://www.youtube.com/playlist?list=PLz8Iz-Fnk_eTpvd49Sa77NiF8Uqq5Iykx) series or really anything in the [docs](/docs/).

There's tons of overlap - _why_ points to _how_, and *how *points to _why_. You could argue we've spent more time on (2) than (1), but the reality is that _we've only scratched the surface of what's possible_.

> ### Principle 3: because Gatsby is so powerful, the ground to cover with writing is huge, yet essential

Reviewer: ```suggestion
## Principle 3: because Gatsby is so powerful, the ground to cover with writing is huge, yet essential
```

- We identified about 50 key messages relating Gatsby to _high-level benefits_ (such as improving security, reducing cost, simplifying your stack), to _different technologies_ (Wordpress, Drupal), _vendors_ (Contentful, AWS), and _communities_(JAMStack, modern development, accessibility). In addition, they cover messages that are important for the wide range of personas in the Gatsby ecosystem -- marketers, educators, open-source contributors, plugin authors, numerous flavors of developers, etc.

Each of these 70-80 messages & workflows probably has ~10 sub-workflows/sub-messages/features — imagine for each of these we had 3-5 case studies, docs pages, dedicated tutorials, starters/themes with annotated open-source source code -- that's thousands, probably tens of thousands, of pieces of content. _That's why we've only scratched the surface of what's possible._

> _(Now imagine a UI on top and the ability to mix & match!)_

Reviewer: What is this referring to? Would love to hear more!

Author: Deliberately being a bit cryptic here :) this could be eg a way to track where you are in a tutorial as well as auto-select and combine starters / themes programatically via UI. But don't want to go too far here as might be a bit distracting.

Reviewer: I personally think it would be less distracting without it, especially since it alludes to something that doesn't exist yet. What if we don't end up solving for that exact thing?

Author: Reasonable!

Our ideal version of Gatsby is a hyperconnected city — connected to all of these somewhat disparate communities (Wordpress, Drupal, modern frontend, accessibility), and so on.

People from each of the communities who have discovered Gatsby are the experts on delivering the message of "why Gatsby" to that community.

> Our objective is **covering the ground** — or if you prefer, _breadth-first search_ rather than _depth-first search_ — making sure that each meaningful message has at least some basic material (both documentation of _how_ and the _why_ of case studies and success stories).

Reviewer: love it 🤓

Author: :)

> In our content database, which lives in Airtable, we store an auto-updating column in the Messages table, which tracks the # of content pieces per message.

Reviewer: Good reminder, thank you!

But...there's a lot of ground to cover? **Yes, there is.**

It's essential to our mission, though, because in order to become the default choice for building on the web, people need to realize that _Gatsby can do anything you conceivably want_ - this is why Wordpress became 30% of the internet! - and we emphasize that by writing about all use cases.

_"Gatsby can't do X"_ gets rebutted by _"Here's a case study of Gatsby doing X"_.

> (For more background on this concept, read about ["economies of scope"](https://www.ribbonfarm.com/2012/10/15/economies-of-scale-economies-of-scope/) or think about companies that have flourished by providing access to literally everything their customers needed — the 1897 Sears Roebuck catalogue with [73K items](https://wpengine.com/reduce-reuse-refresh/); today, Amazon (the "Everything Store") with [100s of millions of items](https://wpengine.com/reduce-reuse-refresh/).)

Reviewer: > [73K items](https://wpengine.com/reduce-reuse-refresh/)
> [100s of millions of items](https://wpengine.com/reduce-reuse-refresh/)

I sort-of expected different links from these two figures. Can they be combined?

Author: Good point.

The challenge, is of course _how do we write all of that content?_

> ### Principle 4: Leverage ourselves through collaboration.

Reviewer: ```suggestion
## Principle 4: Leverage ourselves through collaboration.
```

Some examples:

> - As an Inkteam, we're prioritizing podcasts to conference talks, because they take way less time and build on an existing community/listener base. The exception is key events like JAMStack and DrupalCon that are landmark events for specific communities.

Reviewer: This minimizes the impact of some other big events we have going on this year, as well as speaking contributions of Inkteam members (Amberley, Dustin, Kyle, Preston, Sid, Ward, and anyone else who is presenting). For example, in 2019 there will be Frontend Masters workshops from two Gatsby team members (Jason and myself) and presentations at other high-profile conferences like An Event Apart. Is there a way to communicate priorities while also taking pride in the skills and growing industry reputations of our team members? How about something about making the most of each event to maximize awareness of Gatsby, with workshops, interviews, and Q&A?

Author: That's reasonable, and hints at a better way of phrasing this same concept -- because certainly _don't_ want to downplay that -- it's incredible work! Thanks for the suggestion.

- We encouraging the community to give conference talks, write on their own blogs, and on the Gatsby blog. Individuals can establish themselves publicly as talented developers and Gatsby experts, while companies using Gatsby

- Jason's livestreams are fundamentally collaborative — inviting one person who is an expert in a different technology for cross-pollination.

> - We maintain close relationships with educators who create comprehensive, complemenrary content, and provide encouragement to first-time users that _you can do this!_.

Reviewer: ```suggestion
- We maintain close relationships with educators who create comprehensive, complementary content, and provide encouragement to first-time users that _you can do this!_.
```

If you're doing interesting, innovative things with Gatsby, especially in a professional context, we'd love you to write blog posts, tutorials, and documentation about what you're doing!

> ### Principle 5: Content is not just individual pieces of work, it's compilations.

Reviewer: ```suggestion
## Principle 5: Content is not just individual pieces of work, it's compilations.
```

> If content were music, it isn't just "songs" -- it's "playlists", "albums", "discographies", "soundtracks", and "multi-box sets".

Reviewer: ```suggestion
If content were music, it wouldn't consist of just "songs" -- it would also include "playlists", "albums", "discographies", "soundtracks", and "multi-box sets".
```

> On the one hand, there is _so much left to write_. On the other hand, because we've already been writing so much for so long, we have SO MUCH CONTENT that in many ways we don't know what to do with it.

Author: Thanks! I agree this section is too long, I believe that was due to slightly fuzzy thinking on my part. I've edited and shortened it up a bit.

In the docs, things are structured hierarchically, so usually content is mostly adjacent to related content on the left sidebar.

But outside the docs, that's far from the case. We have [tags on the blog](https://www.gatsbyjs.org/blog/tags/), which we use occasionally, but not really very well.

But there's [all sorts of other content types] — [using-\* example sites / guides](https://github.com/gatsbyjs/gatsby/tree/master/examples), [starters](/starters/), conference talks, podcast appearances, tweets, and whatever Gatsby community members dream up and publish!

**It's a constant challenge to organize these in ways that are useful and obvious for the end-user**. We're always attacking this in five or ten different ways -- discoverability via search, cross-linking, hierarchy, navbars and breadcrumbs, turning conference talks into blog posts and blog posts into docs...

_Why does this matter?_ We want to make it easy for the user, so that if and when they want to deep dive, it's easy for them to go read ten or twenty articles on their topic of choice.

> ### Principle 6: make it easy for folks to become engrossed with Gatsby

Reviewer: ```suggestion
## Principle 6: make it easy for folks to become engrossed with Gatsby
```
